src/exec/lib.rs

```diff
@@ -72,15 +72,18 @@ pub trait ExecutorFactory
 pub struct NamedSchema<'a> {
   pub names: &'a [&'a str],
   pub types: &'a [&'a Ty],
+  pub enc_types: &'a [&'a EncType],
 }
 
 impl<'a> NamedSchema<'a> {
-  pub fn new(names: &'a [&'a str], types: &'a [&Ty]) -> NamedSchema<'a> {
+  pub fn new(names: &'a [&'a str], types: &'a [&Ty], enc_types: &'a [&EncType]) -> NamedSchema<'a> {
```

Comment:
I have a plan to separate some logical description like schema and some physical description. But, there is no implementation for the physical description. So, now is fine. I'll refactor them :)